.github/workflows/ci.yml

```diff
@@ -39,4 +43,29 @@ jobs:
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           BRANCH_NAME: ${{ github.ref }}
-        run: if [[ -n $SONAR_TOKEN ]]; then ./housekeeping/sonar.sh; else echo "Some secret undefined. Step passed..."; fi
\ No newline at end of file
+        run: if [[ -n $SONAR_TOKEN ]]; then ./housekeeping/sonar.sh; else echo "Some secret undefined. Step passed..."; fi
+      - name: Publish lib to nexus for branch
```

Comment:
I suspect two steps below can be merged to a one with some processing logic, not sure what the best practice is

Reply:
There are two different conditions for branch and for tag with different ways to set version and no "else" option